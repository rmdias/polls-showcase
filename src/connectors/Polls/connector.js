import _ from 'lodash'
import { connect } from 'react-redux'
import { polls } from 'actions/polls'

export default connect(
  (state, ownProps) => ({
    pollsList: _.get(state, 'polls.list', []),
    loading: _.get(state, 'polls.loading', true)
  }),
  (dispatch) => ({
    onFetchPolls: page => dispatch(polls.fetch.request())
  })
)
