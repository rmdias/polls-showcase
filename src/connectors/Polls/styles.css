.polls {
  padding-top: 40px;
}

.polls_items {
  display: flex;
  flex-wrap: wrap;
}

.polls__loader {
  margin: 0 auto;
}
