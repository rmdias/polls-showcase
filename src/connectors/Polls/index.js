import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import InfiniteScroll from 'react-infinite-scroller'

import { CircularLoader, Button } from 'luna-ui-lib'

import NoResults from 'components/NoResults'
import PollItem from 'components/PollItem'

import './styles.css'
import connector from './connector'

class Polls extends PureComponent {
  constructor(props) {
    super(props)

    window.scrollTo(0, 0)
  }

  static propTypes = {
    hasMore: PropTypes.bool,
    loading: PropTypes.bool,
    pollsList: PropTypes.array,
    onFetchPolls: PropTypes.func
  }

  componentDidMount() {
    this.props.onFetchPolls(1)
  }

  render() {
    const showLoader = this.props.loading && !this.props.pollsList.length
    const showNoResults = !this.props.loading && !this.props.pollsList.length
    const loader = <CircularLoader className="polls__loader" key={0} kind="primary" size={50} border={1} />
    const Items = this.props.pollsList.map(poll => {
      return <PollItem key={JSON.stringify(poll)} poll={poll}></PollItem>
    })

    return (
      <div className="polls">

        <Link to="/new" className="poll__create_new">
          <Button size="large" kind="primary">
            Create
          </Button>
        </Link>
        <h1>Questions</h1>
        { showLoader && loader }
        { showNoResults && <NoResults /> }

        <div className="polls_items">
          { Items }
          <InfiniteScroll
            pageStart={1}
            loadMore={(e) => this.props.onFetchPolls(e)}
            hasMore={this.props.hasMore}
            loader={loader}
          >
            { Items }
          </InfiniteScroll>
        </div>
      </div>
    )
  }
}

export default connector(Polls)
