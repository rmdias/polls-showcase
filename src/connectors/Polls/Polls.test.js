import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { BrowserRouter as Router } from 'react-router-dom'

const middlewares = []
const mockStore = configureStore(middlewares)

import Polls from './index'
import PollsMock from './__mocks__/__mock__polls'

import { Theme } from 'luna-ui-lib'

let initialState
const polls = {
  list: [],
  loading: true
}

beforeEach(() => {
  initialState = { polls }
  global.scrollTo = jest.fn()
})

describe('<Polls />', () => {
  it('toMatchSnapshot: Loading', () => {
    const store = mockStore(initialState)
    const component = (
      <Theme>
        <Router>
          <Polls store={store}/>
        </Router>
      </Theme>
    )

    const tree = renderer
      .create(component)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('toMatchSnapshot: Load <Polls /> as list of <PollItem />', () => {
    initialState.polls.list = PollsMock
    const store = mockStore(initialState)
    const component = (
      <Theme>
        <Router>
          <Polls store={store}/>
        </Router>
      </Theme>
    )
    const tree = renderer
      .create(component)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
