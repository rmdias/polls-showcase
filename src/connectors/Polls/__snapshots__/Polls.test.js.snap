// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Polls /> toMatchSnapshot: Load <Polls /> as list of <PollItem /> 1`] = `
<div>
  <div
    className="polls"
  >
    <h1>
      Questions
    </h1>
    <div
      className="polls_items"
    >
      <a
        className="poll__item"
        href="/questions/10"
        onClick={[Function]}
      >
        <div
          className="poll__item__area sc-dxgOiQ kRJcgL"
        >
          <h4
            className="poll__item__title"
          >
            <strong>
              Game Genre
            </strong>
          </h4>
          <span
            className="sc-ckVGcZ GYKlq"
          >
            Published at 
            27.05.2015 11:22:26
          </span>
          <p>
            3
             choices
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`<Polls /> toMatchSnapshot: Loading 1`] = `
<div>
  <div
    className="polls"
  >
    <h1>
      Questions
    </h1>
    <div
      className="polls__loader sc-kpOJdX jIxhBs"
      kind="primary"
      size={50}
    />
    <div
      className="polls_items"
    />
  </div>
</div>
`;
